clusterctl/main_integration_test.go

```diff
@@ -0,0 +1,174 @@
+// +build integration
+
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main_test
+
+import (
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path"
+	"strings"
+	"syscall"
+	"testing"
+)
+
+const (
+	clusterctlBinaryName = "clusterctl"
+	workingDirectoryName = "clusterctl"
+)
+
+// run these tests with the flag "-update" to update the values stored in all of the golden files
+var update = flag.Bool("update", false, "update .golden files")
+
+func TestMain(m *testing.M) {
+	flag.Parse()
+	err := setupPrerequisites()
+	if err != nil {
+		fmt.Printf("unable to setup prerequisites: %v\n", err)
+		os.Exit(1)
+	}
+	os.Exit(m.Run())
+}
+
+func TestEmptyAndInvalidArgs(t *testing.T) {
+	testCases := []struct {
+		name            string
+		args            []string
+		exitCode        int
+		fixtureFilename string
+	}{
+		{"no arguments", []string{}, 0, "no-args.golden"},
+		{"no arguments with invalid flag", []string{"--invalid-flag"}, 1, "no-args-invalid-flag.golden"},
+		{"create with no arguments", []string{"create"}, 0, "create-no-args.golden"},
+		{"create with no arguments with invalid flag", []string{"create", "--invalid-flag"}, 1, "create-no-args-invalid-flag.golden"},
+		{"create cluster with no arguments", []string{"create", "cluster"}, 1, "create-cluster-no-args.golden"},
+		{"create cluster with no arguments with invalid flag", []string{"create", "cluster", "--invalid-flag"}, 1, "create-cluster-no-args-invalid-flag.golden"},
+		{"delete with no arguments", []string{"delete"}, 0, "delete-no-args.golden"},
+		{"delete with no arguments with invalid flag", []string{"delete", "--invalid-flag"}, 1, "delete-no-args-invalid-flag.golden"},
+		{"delete cluster with no arguments", []string{"delete", "delete"}, 1, "delete-cluster-no-args.golden"},
+		{"delete cluster with no arguments with invalid flag", []string{"delete", "delete", "--invalid-flag"}, 1, "delete-cluster-no-args-invalid-flag.golden"},
```

Comment:
`[]string{"delete", "delete", ...}`

Reply:
This is necessary due to a bug in the CLI. I will follow up with a fix.